src/util.rs

```diff
@@ -69,19 +70,35 @@ pub extern "C" fn callback(callback_info: v8::FunctionCallbackInfoPtr_Value) {
         let args = (0..length)
             .map(|i| value::Value::from_raw(&isolate, *callback_info.Args.offset(i)))
             .collect();
-        let info = value::FunctionCallbackInfo {
-            isolate: isolate.clone(),
-            length: length,
-            args: args,
-            this: value::Object::from_raw(&isolate, callback_info.This),
-            holder: value::Object::from_raw(&isolate, callback_info.Holder),
-            new_target: value::Value::from_raw(&isolate, callback_info.NewTarget),
-            is_construct_call: 0 != callback_info.IsConstructCall,
-        };
+
 
         let result = panic::catch_unwind(|| {
+            let holder = value::Object::from_raw(&isolate, callback_info.Holder);
+
+            let internal = if holder.internal_field_count() > 0 {
+                if let Some(external) = holder.get_internal_field(0).into_external() {
```

Comment:
I can't recall anymore how this gets converted from the C++ side. Object::get_internal_field will return a null pointer if the space has been allocated but no instance has been set. And a Local<Value> that holds a null pointer will result in IsEmpty() returning true. To err on the safe side, Object::get_internal_field could return Option<Value> so that if the Object_GetInternalField call returns null, that case can return None and then I can change the callback code to only execute in the Some case.